### PR TITLE
bug when creating depth campaign

### PR DESCRIPTION
**Pull request by JosuaKrause**
- [x] await the result first
- [x] keep one AND when removing clauses and there are more clauses after
- [x] don't try to produce ogp for pads that don't have the information needed

report here: #82

### CODE REVIEW

File: routes/browse/pads/load/statistics.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ module.exports = async kwargs => {
 				$1:raw
 			GROUP BY p.status
 			ORDER BY p.status
-		;`, [ full_filters.replace(/(AND\s)?p\.status IN \([\'\d\,\s]+\)(\sAND\s)?/g, '') ]).then(d => { return { persistent: d } })
+		;`, [ full_filters.replace(/(AND\s)?p\.status IN \([\'\d\,\s]+\)(\sAND\s)?/g, '$2') ]).then(d => { return { persistent: d } })
```
Comment (JosuaKrause) [inline routes/browse/pads/load/statistics.js:57]: we want to keep the second `AND` if it exists

File: routes/contribute/pad/index.js [hunk 2]
```diff
@@ -221,9 +221,9 @@ module.exports = (req, res) => {
 
 					if (id) data = await datastructures.legacy.publishablepad({ connection: t, data });
 
-					const excerpt = data.status > 2 ? { title: data.title, txt: parsers.getTxt(data)[0], img: { src: parsers.getImg(data)[0], width: 300, height: 200 }, p: true } : null
+					const excerpt = (data && data.status && data.status > 2) ? { title: data.title, txt: parsers.getTxt(data)[0], img: { src: parsers.getImg(data)[0], width: 300, height: 200 }, p: true } : null
```
Comment (JosuaKrause) [inline routes/contribute/pad/index.js:224]: `data` here can be uninitialized in certain circumstances (e.g., a new pad) -- we can make excerpt null if data happens to not have any information

File: routes/mobilize/cohort/index.js [hunk 3]
```diff
@@ -41,10 +41,10 @@ module.exports = (req, res) => {
 				}).catch(err => console.log(err)))
 			} else if (pinboard) {
 				batch.push(ownDB().then(async ownId => {
-					const padlist = await DB.general.any(`
+					const padlist = (await DB.general.any(`
 						SELECT pc.pad FROM pinboard_contributions pc
 						WHERE pc.pinboard = $1 AND pc.db = $2
-					;`, [ pinboard, ownId ]).map((row) => row.pad);
+					;`, [ pinboard, ownId ])).map((row) => row.pad);
```
Comment (JosuaKrause) [inline routes/mobilize/cohort/index.js:47]: missing (...) making `map` trying to work on the promise instead of *awaiting* the result

**Comment by JosuaKrause** [comment]
should be gtm

**Review by myjyby — APPROVED** [review]
simple fixes. all looks good thanks.

**Pull request merged**